landlab/components/landslides/landslide_probability.py

```diff
@@ -641,6 +824,23 @@ def _interpolate_HSD_dict(self):
 
         self._interpolated_HSD_dict = HSD_dict
 
+    def _interpolate_HSD_array(self):
+        """Method to extrapolate input data.
+
+        This method uses a non-parametric approach to expand the input
+        depth to water table array to the length of number of iterations. Output is
+        a new array of interpolated values for each HSD id.
```

Comment:
Might add a few more sentences on what the nonparametric approach is. This is a private function, so low priority recommendation.